Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,946 @@
+                    }
+                }
+                else if (state == 3)
+                {
+                    if (c == ';')
+                    {
+                        if (number == 0)
+                        {
+                            output.Append(rawEntity.ToString() + ";");
+                        }
+                        else if (number > 65535)
+                        {
+                            output.Append("&#");
+                            output.Append(number.ToString(Helpers.InvariantCulture));
+                            output.Append(";");
+                        }
+                        else
+                        {
+                            output.Append((char)number);
+                        }
+
+                        state = 0;
+                        entity.Length = 0;
+                        rawEntity.Length = 0;
+                        have_trailing_digits = false;
+                    }
+                    else if (is_hex_value && Uri.IsHexDigit(c))
+                    {
+                        number = number * 16 + Uri.FromHex(c);
+                        have_trailing_digits = true;
+                        rawEntity.Append(c);
+                    }
+                    else if (Char.IsDigit(c))
+                    {
+                        number = number * 10 + ((int)c - '0');
+                        have_trailing_digits = true;
+                        rawEntity.Append(c);
+                    }
+                    else if (number == 0 && (c == 'x' || c == 'X'))
+                    {
+                        is_hex_value = true;
+                        rawEntity.Append(c);
+                    }
+                    else
+                    {
+                        state = 2;
+                        if (have_trailing_digits)
+                        {
+                            entity.Append(number.ToString(Helpers.InvariantCulture));
+                            have_trailing_digits = false;
+                        }
+
+                        entity.Append(c);
+                    }
+                }
+            }
+
+            if (entity.Length > 0)
+            {
+                output.Append(entity.ToString());
+            }
+            else if (have_trailing_digits)
+            {
+                output.Append(number.ToString(Helpers.InvariantCulture));
+            }
+
+            return output.ToString();
+        }
+
+        internal static bool NotEncoded(char c)
+        {
+            return (c == '!' || c == '(' || c == ')' || c == '*' || c == '-' ||
+                c == '.' || c == '_');
+        }
+
+        internal static void UrlEncodeChar(char c, Stream result, bool isUnicode)
+        {
+            if (c > 255)
+            {
+                //FIXME: what happens when there is an internal error?
+                //if (!isUnicode)
+                //	throw new ArgumentOutOfRangeException ("c", c, "c must be less than 256");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)